File: indexter-games/src/main/java/com/github/ykrasik/indexter/games/datamodel/persistence/Library.java
package com.github.ykrasik.indexter.games.datamodel.persistence;

import com.github.ykrasik.indexter.games.datamodel.GamePlatform;
import com.github.ykrasik.indexter.id.Id;
import lombok.NonNull;
import lombok.Value;

import java.nio.file.Path;

/**
 * @author Yevgeny Krasik
 */
@Value
public class Library {
    @NonNull private final Id<Library> id;
    @NonNull private final String name;
    @NonNull private final Path path;
    @NonNull private final GamePlatform platform;
}

File: gamedex-datamodel/src/main/kotlin/com/github/ykrasik/gamedex/datamodel/Library.kt
package com.github.ykrasik.gamedex.datamodel

import java.io.File

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 11:29
 */
data class Library(
    val id: Int,
    val data: LibraryData
) : Comparable<Library> {

    val name: String get() = data.name
    val path: File get() = data.path
    val platform: GamePlatform get() = data.platform

    override fun compareTo(other: Library) = data.compareTo(other.data)
}

data class LibraryData(
    val path: File,
    val name: String,
    val platform: GamePlatform
) : Comparable<LibraryData> {

    override fun compareTo(other: LibraryData) = name.compareTo(other.name)
}